### PR TITLE
Updated rules regarding hostname in ingresstrait

### PR DESCRIPTION
**Pull request by sk2701**
This PR updates the rules for hostname in ingress-trait

### CODE REVIEW

File: content/en/docs/reference/API/OAM/IngressTrait.md [hunk 1]
```diff
@@ -108,6 +107,12 @@ spec:
                                     - "customer"
 
 ```
+With reference to the hostname, the following rules need to be followed:
+
+1. If you are providing a hostname, you can provide a certificate otherwise verrazzano will generate one for you.
+2. If you are providing a certifcate, then you must provide a hostname.
+3. If you do not provide either a hostname or a certificate, verrazzano will generate one for you.
+
```
Comment (MelwynPaul) [inline content/en/docs/reference/API/OAM/IngressTrait.md:115]: The following are the rules related to the hostname: 

- If you provide a hostname, then you have an option to provide a certificate. If you do not provide a certificate Verrazzano generates one for you.
- If you provide a certificate, then you must provide a hostname.
- If you do not provide either a hostname or a certificate, then Verrazano automatically generates them for you.

(unordered list please)

**Review by MelwynPaul — COMMENTED** [review]
Added a few comments. 
Thank you.

File: content/en/docs/reference/API/OAM/IngressTrait.md [hunk 2]
```diff
@@ -108,6 +107,12 @@ spec:
                                     - "customer"
 
 ```
+The following are the rules related to the hostname:
+
+- If you provide a hostname, then you have an option to provide a certificate. If you do not provide a certificate Verrazzano generates one for you.
```
Comment (rosemarymarano) [inline content/en/docs/reference/API/OAM/IngressTrait.md:112]: If you do not provide a certificate Verrazzano generates one for you -> If you do not provide a certificate, then Verrazzano generates one for you.

File: content/en/docs/reference/API/OAM/IngressTrait.md [hunk 3]
```diff
@@ -108,6 +107,12 @@ spec:
                                     - "customer"
 
 ```
+The following are the rules related to the hostname:
```
Comment (rosemarymarano) [inline content/en/docs/reference/API/OAM/IngressTrait.md:110]: The following are the rules related to the hostname: -> Use the following rules related to the host name:

File: content/en/docs/reference/API/OAM/IngressTrait.md [hunk 4]
```diff
@@ -108,6 +107,12 @@ spec:
...
+
+- If you provide a hostname, then you have an option to provide a certificate. If you do not provide a certificate Verrazzano generates one for you.
+- If you provide a certificate, then you must provide a hostname.
```
Comment (rosemarymarano) [inline content/en/docs/reference/API/OAM/IngressTrait.md:113]: hostname -> host name (two words; please correct globally)

File: content/en/docs/reference/API/OAM/IngressTrait.md [hunk 5]
```diff
@@ -108,6 +107,12 @@ spec:
...
+- If you do not provide either a hostname or a certificate, then Verrazzano automatically generates them for you.
```
Comment (rosemarymarano) [inline content/en/docs/reference/API/OAM/IngressTrait.md:114]: then Verrazzano automatically generates them for you. -> then Verrazzano generates them for you.

**Review by rosemarymarano — COMMENTED** [review]
Few edits, please incorporate them.

**Review by rosemarymarano — APPROVED** [review]
LGTM

**Review by MelwynPaul — APPROVED** [review]
Thank you.

**Pull request merged**